### ISSUE TITLE
Python meterpreter no longer works with macOS 10.12 system python installation

### ISSUE THREAD

**Issue by bcook-r7** [rapid7/metasploit-framework#7367]
## Steps to reproduce

From @mubix on IRC:

python meterpreter fails with:

```
System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000110
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]
```

**Comment by zeroSteiner** [comment]
I'm going to take a look at this. I'm guessing it's related to rapid7/metasploit-payloads#144 but I'll see if I can reproduce it.

**Comment by busterb** [comment]
You're probably right!

**Comment by kaospunk** [comment]
I tested this on macOS 10.12.3 using the native system python as well as a version from brew. I tested using the latest version of metasploit from git:
Framework Version: 4.13.21-dev-cba5e26

**The latest version works properly for me which both python installs in that sessions are established and I can interact with the session with no issue.**

To test the older versions of the python meterpreter I dropped the two files from https://github.com/rapid7/metasploit-payloads/tree/09a955924438c5c3bea08021551c77888684c43c/python/meterpreter in to ~/.msf4/payloads/meterpreter. After doing so the results with either install of python are the same. A session appears to be established but it is invalid so it is torn down. 

I was not able to reproduce the crash message provided by @mubix above though in any of my testing.
